Client/Client.Core/Client.Core.Static/Client.cpp

```diff
@@ -22,149 +24,286 @@ bool Client::connect(const std::string_view& host, const uint16_t port)
+    // Temporarily commented out function
+    // send(message);
+}
+
+void Client::loop()
+{
+    while (!_incomingMessagesQueue.empty())
+    {
+        const Message message = _incomingMessagesQueue.pop_front();
+        std::string   output  = "[" + std::to_string(message.mHeader.mTimestamp.time_since_epoch().count()) + "]\n";
+        std::cout << output;
+        switch (message.mHeader.mMessageType)
+        {
+            case MessageType::LoginAnswer:
+            {
+                auto loginSuccessful = std::any_cast<bool>(message.mBody);
+                onLoginAnswer(loginSuccessful);
+            }
+            break;
+
+            case MessageType::ServerAccept:
+            {
+                onServerAccepted();
+            }
+            break;
+
+            case MessageType::ServerPing:
+            {
+                std::chrono::system_clock::time_point timeNow = std::chrono::system_clock::now();
+                std::chrono::system_clock::time_point timeThen;
+                timeThen = message.mHeader.mTimestamp;
+                onServerPing(std::chrono::duration<double>(timeNow - timeThen).count());
+            }
+            break;
+
+            case MessageType::ServerMessage:
+            {
+                // TODO: add handling
+                uint64_t clientID = 0;
+                onServerMessage(clientID);
+            }
+            break;
+
+            case MessageType::ChannelListRequest:
+            {
+                auto channels = std::any_cast<std::vector<Network::ChannelInfo>>(message.mBody);
+                onChannelListRequest(channels);
+            }
+            break;
+
+            case MessageType::MessageHistoryAnswer:
+            {
+                auto messages = std::any_cast<std::vector<Network::MessageInfo>>(message.mBody);
+                onMessageHistoryAnswer(messages);
+            }
+            break;
+
+            case MessageType::MessageStoreAnswer:
+            {
+                auto code = std::any_cast<Utility::StoringMessageCodes>(message.mBody);
+                onMessageStoreAnswer(code);
+            }
+            break;
+
+            case MessageType::RegistrationAnswer:
+            {
+                auto code = std::any_cast<Utility::RegistrationCodes>(message.mBody);
+
+                onRegistrationAnswer(code);
+            }
+            break;
+
+            case MessageType::UserMessageDeleteAnswer:
+            {
+                onUserMessageDeleteAnswer();
+                // Temporarily commented out code
+                // auto messageInfo = std::any_cast<Network::MessageDeletedInfo>(message.mBody);
+            }
+            break;
+
+            default:
+                std::cerr << "[Client][Warning] unimplemented[" << uint32_t(message.mHeader.mMessageType) << "]\n";
+        }
+    }
+}
+
+void Client::onLoginAnswer(bool success)
+{
+    (void)(success);
+    std::cerr << "[Client][Warning] login answer is not implemented\n";
```

Comment:
In such case, you should use Utility::exceptions 'NotImplementedException'.

Reply:
Unimplemented handlers is not a critical error to use exceptions